File: Bit Manipulation/Hamming-Distance.cpp
int solve(int first, int second) {
  int hamming_distance = 0;
  while (first || second) {
    if (first & 1 != second & 1) {  // numbers are of different parity, last bits are different
      ++hamming_distance;
    }
    first >>= 1; second >>= 1;  // remove last bits
  }
  return hamming_distance;
}

File: Bit Manipulation/Hamming-Distance.py
class Solution:
  def solve(self, x, y):
    hamming_distance = 0
    while x or y:
      if x & 1 != y & 1:  # x and y are of different parity, last bits are different
        hamming_distance += 1
      x >>= 1; y >>= 1  # remove last bits
    return hamming_distance
